src/main/kotlin/com/terraformation/backend/admin/AdminCohortsController.kt

```diff
@@ -163,7 +170,123 @@ class AdminCohortsController(
     return redirectToCohort(cohortId)
   }
 
+  @PostMapping("/cohorts/{cohortId}/deliverables")
+  fun deliverablesRedirect(
+      model: Model,
+      @PathVariable cohortId: CohortId,
+      @RequestParam deliverableId: DeliverableId,
+  ): String = redirectToCohortDeliverable(cohortId, deliverableId)
+
+  @GetMapping("/cohorts/{cohortId}/deliverables/{deliverableId}")
+  fun viewDeliverable(
+      model: Model,
+      @PathVariable cohortId: CohortId,
+      @PathVariable deliverableId: DeliverableId,
+      redirectAttributes: RedirectAttributes
+  ): String {
+    requirePermissions { readAllDeliverables() }
+
+    val cohort =
+        try {
+          cohortStore.fetchOneById(cohortId, CohortDepth.Cohort, CohortModuleDepth.Module)
+        } catch (e: CohortNotFoundException) {
+          log.warn("Cohort not found", e)
+          redirectAttributes.failureMessage = "Cohort not found: ${e.message}"
+          return redirectToCohortHome()
+        }
+
+    val dueDateModel =
+        try {
+          deliverableDueDateStore
+              .fetchDeliverableDueDates(cohortId = cohortId, deliverableId = deliverableId)
+              .firstOrNull()
+        } catch (e: Exception) {
+          redirectAttributes.failureMessage = "Error looking up deliverable due date: ${e.message}"
+          return redirectToCohort(cohortId)
+        }
+
+    if (dueDateModel == null) {
+      redirectAttributes.failureMessage =
+          "Deliverable $deliverableId not assocaited with cohort $cohortId"
+      return redirectToCohort(cohortId)
+    }
+
+    val module =
+        try {
+          moduleStore.fetchOneById(dueDateModel.moduleId)
+        } catch (e: ModuleNotFoundException) {
+          redirectAttributes.failureMessage = "Module for deliverable not found: ${e.message}"
+          return redirectToCohort(cohortId)
+        }
+
+    val cohortProjects = module.cohorts.first { it.cohortId == cohortId }.projects!!
+
+    val cohortModule = cohort.modules.first { it.moduleId == module.id }
+    val deliverable = module.deliverables.first { it.id == deliverableId }
+
+    val submissions =
+        deliverableStore
+            .fetchDeliverableSubmissions(deliverableId = deliverableId)
+            .filter { cohortProjects.contains(it.projectId) }
+            .sortedBy { it.projectId.value }
+
+    model.addAttribute("cohort", cohort)
+    model.addAttribute("cohortModule", cohortModule)
+    model.addAttribute("deliverable", deliverable)
+    model.addAttribute("dueDates", dueDateModel)
+    model.addAttribute("module", module)
+    model.addAttribute("submissions", submissions)
+    model.addAttribute("canManageDeliverables", currentUser().canManageDeliverables())
```

Comment:
The rest of the attributes are alphabetized, so may as well put this one at the top.